crates/ra_hir/src/ty/tests.rs

```diff
@@ -772,60 +774,89 @@ fn test(x: &i32) {
 
     let lambda = |a: u64, b, c: i32| { a + b; c };
 
+    let A(n) = &A(1);
+    let A(n) = &mut A(1);
+
+    let v = &(1, &2);
+    let (_, &w) = v;
```

Comment:
This tests the [`else if let Pat::Ref { .. }`](https://github.com/rust-analyzer/rust-analyzer/pull/982/files#diff-680da4262b0a53439ea60a27e00f9494R615), which is rust-lang/rust#46688.

Reply:
maybe a separate test for that, plus a marker (`tested_by!`), would be nice :)

Reply:
Also, in general, I'd prefer adding the new test cases in a separate test -- it's already not so easy what's what here

Reply:
I hadn't come across `tested_by!`. Very neat! :slightly_smiling_face: 

I separated this out into two test-cases, as I think it makes the snapshots easier to read, and makes it clear what is being tested.